helpers/mathJax.js

```diff
@@ -0,0 +1,124 @@
+				const startup = window.MathJax.startup;
+
+				//
+				//  Extend HTMLAdaptor to handle shadowDOM as the document
+				//
+				class ShadowAdaptor extends HTMLAdaptor {
+					body(doc) {
+						return doc.body || (doc.firstChild || {}).lastChild || doc;
+					}
+					create(kind, ns) {
+						const document = (this.document.createElement ? this.document : this.window.document);
+						return (ns ?
+							document.createElementNS(ns, kind) :
+							document.createElement(kind));
+					}
+					head(doc) {
+						return doc.head || (doc.firstChild || {}).firstChild || doc;
+					}
+					root(doc) {
+						return doc.documentElement || doc.firstChild || doc;
+					}
+					text(text) {
+						const document = (this.document.createTextNode ? this.document : this.window.document);
+						return document.createTextNode(text);
+					}
+				}
+
+				//
+				//  Extend HTMLHandler to handle shadowDOM as document
+				//
+				class ShadowHandler extends HTMLHandler {
+					create(document, options) {
+						const adaptor = this.adaptor;
+						if (typeof(document) === 'string') {
+							document = adaptor.parse(document, 'text/html');
+						} else if ((document instanceof adaptor.window.HTMLElement || document instanceof adaptor.window.DocumentFragment) && !(document instanceof window.ShadowRoot)) {
+							const child = document;
+							document = adaptor.parse('', 'text/html');
+							adaptor.append(adaptor.body(document), child);
+						}
+						//
+						//  We can't use super.create() here, since that doesn't
+						//  handle shadowDOM correctly, so call HTMLHandler's parent class
+						//  directly instead.
+						//
+						return AbstractHandler.create.call(this, document, options);
+					}
+				}
+
+				//
+				//  Register the new handler and adaptor
+				//
+				startup.registerConstructor('HTMLHandler', ShadowHandler);
+				startup.registerConstructor('browserAdaptor', () => new ShadowAdaptor(window));
+
+				//
+				//  A service function that creates a new MathDocument from the
+				//  shadow root with the configured input and output jax, and then
+				//  renders the document.  The MathDocument is returned in case
+				//  you need to rerender the shadowRoot later.
+				//
+				window.MathJax.typesetShadow = function(root) {
+					const InputJax = startup.getInputJax();
+					const OutputJax = startup.getOutputJax();
+					const html = mathjax.document(root, { InputJax, OutputJax });
+					html.render();
+					return html;
+				};
+
+				//
+				//  Now do the usual startup now that the extensions are in place
+				//
+				window.MathJax.startup.defaultReady();
+			}
+		}
+	};
+
+	mathJaxLoaded = new Promise(resolve => {
+		const script = document.createElement('script');
+		script.async = 'async';
+		script.onload = resolve;
+		script.src = 'https://s.brightspace.com/lib/mathjax/3.1.2/mml-chtml.js';
+		document.head.appendChild(script);
+	});
+
+	return mathJaxLoaded;
+
+}
+
+window.D2L = window.D2L || {};
```

Comment:
This is the only bit that's added here.  This is needed to expose the loader, for example in the context of Content Topics.

Another option I am contemplating is having the consumer use a dynamic import instead.  For example:
```javascript
import('..the-location-of/mathJax.js').then(module => {
    module.loadMathJax();
});
```
But then it requires the consumer to use a dynamic import, which I'm not sure if we'd encounter some FOUC.

Reply:
Aside, this would be a technique that _could_ be used to use some of our other helpers in the monolith without having to expose them explicitly on the window in BSI.

Reply:
Right, so typically we'd do this bit inside BSI, [like this](https://github.com/Brightspace/brightspace-integration/blob/master/web-components/bsi-unbundled.js#L16). I think I'd prefer we stick with one technique or the other.

Doesn't this approach still require that BSI import it (to make sure it's on the page).?

Reply:
So in this case the BSI option won't be possible since it's not included in Content Topics.

Reply:
But yeah, there will be code in the monolith that wants to call this too.

Reply:
Right... so you'll need to expose this as a module similar to our other components and load that module on the page(s) that need it.

Reply:
Ok, I've done a bunch of testing with this including different browsers, both Dev and Prod configurations of BSI and it seems to be working as expected for Content Topics.  I can move the `window.D2L.MathJax` bit to a module in BSI.